onnxscript/ir/passes/common/unused_removal.py

```diff
@@ -186,3 +186,33 @@ def call(self, model: ir.Model) -> ir.passes.PassResult:
                 modified |= self._process_graph_like(function, used_domains=set())
 
         return ir.passes.PassResult(model, modified=modified)
+
+
+class RemoveUnusedMetadataAndDocStringPass(ir.passes.InPlacePass):
+    def call(self, model: ir.Model) -> ir.passes.PassResult:
+        modified = False
+        # 1. Clean up all of the nodes metadata properties
+        for node in ir.traversal.RecursiveGraphIterator(model.graph):
+            if len(node.metadata_props) > 0:
+                modified = True
+                logger.debug("Removed metadata from %s nodes", node.name)
+            node.metadata_props.clear()
+
+            # TODO: Use model.graphs to get the graph instead of node.graph
+            # 2. Clean up the main graph metadata properties
+            # and doc_string
+            assert node.graph is not None
+            if len(node.graph.metadata_props) > 0 or node.graph.doc_string is not None:
+                modified = True
+                logger.debug("Removed metadata from %s graph", node.graph.name)
+            node.graph.metadata_props.clear()
+            node.graph.doc_string = None
+
+        # 3. Clean up all of the functions metadata properties
+        for function in model.functions.values():
```

Comment:
loop though nodes and subgraphs in a function too?

Reply:
I guess I can use ir.traversal.RecursiveGraphIterator(function)?

Reply:
Yes. Should be the exact same usage.